### DISCUSSION TITLE
Enroll Storage and Switches through Teleport agent

### DISCUSSION THREAD

**Discussion by skarthic15** [gravitational/teleport#39295] [Q&A] [answered]
Is it possible to enroll storage, switches or router devices?
Agents cannot deployed on these devices...Want connect SSH and https also.

**Comment by stevenGravy** [comment, marked as answer]
If the devices support certificates then yes the agentless approach could work. https://goteleport.com/docs/server-access/openssh/ Otherwise TCP app can provide connectivity but not authentication.

**Reply by skarthic15** [reply]
Is is possible to use teleport agent as proxy to register and connect these devices?

teleport auth/proxy --> teleport agent --> switch/storage

**Reply by skarthic15** [reply]
Can you please give more details on "TCP app"?\
ssh connectivity is possible with TCP app?

**Reply by webvictim** [reply]
The Teleport `ssh_service‘ does not currently support being used as a relay/gateway to connect to other devices. This functionality is being investigated for the future - the request is tracked in this issue: https://github.com/gravitational/teleport/issues/28541

You can forward any TCP-based service (like SSH) using Teleport application access if desired. It is easier to configure, but less fully featured - for example, it has no login-based RBAC and no SSH session recording. The documentation can be found here: https://goteleport.com/docs/application-access/guides/tcp
